addons/l10n_anz_ubl_pint/models/account_edi_xml_pint_anz.py

```diff
@@ -106,6 +119,36 @@ def _export_invoice_constraints(self, invoice, vals):
                 if tax_subtotal_val['tax_category_vals']['tax_category_code'] not in ANZ_TAX_CATEGORIES:
                     constraints['sg_vat_category_required'] = _("You must set a tax category on each taxes of the invoice.\nValid categories are: S, E, Z, G, O")
 
+        # There should be at most one tax breakdown of type "Not subject to tax".
+        if len([b for b in vals['taxes_vals']['tax_details'] if b.get('tax_category_id') == 'O']) > 1:
+            if vals['supplier'].vat:
+                constraints['anz_duplicate_tax_breakdown'] = \
+                    self.env._("A tax breakdown of type 'Not subject to tax' should appear at most"
+                               " once in the tax breakdown")
+            else:
+                # If the supplier is not registered for GST, this likely means that the user
+                # is trying to use taxes that are not zero-rated or exempted.
+                constraints['anz_non_gst_supplier_tax_scope'] = \
+                    self.env._("Suppliers not registered for GST cannot use taxes that are not"
+                               " zero-rated. Please ensure the tax category is set to 'O' (Services"
+                               " outside scope of tax) with a tax amount of 0.")
+
+        # Tax category of type "Not subject to tax" must have tax amount 0
+        for tax_category, tax_category_vals in vals['taxes_vals']['tax_details'].items():
+            if tax_category['tax_category_id'] == 'O' \
+               and not float_is_zero(tax_category_vals['tax_amount'], precision_digits=2):
+                if vals['supplier'].vat:
+                    constraints['anz_tax_breakdown_amount'] = \
+                        self.env._("A tax category of type 'Not subject to tax' must have tax"
+                                   " amount set to 0")
+                else:
+                    # If the supplier is not registered for GST, this likely means that the user
+                    # is trying to use taxes that are not zero-rated or exempted.
+                    constraints['anz_non_gst_supplier_tax_scope'] = \
+                        self.env._("Suppliers not registered for GST cannot use taxes that are not"
+                                   " zero-rated. Please ensure the tax category is set to 'O'"
+                                   " (Services outside scope of tax) with a tax amount of 0.")
```

Comment:
Same here, why is the tax amount check only done for vat companies?
Even if the amount should always be 0 for non registered ones, the check can still be done for all.

And this second constraints is also unclear; as you don't really validate that there is only one breadown with the category O but just the amount when the category is O